app/src/full/java/io/homeassistant/companion/android/notifications/MessagingService.kt

```diff
@@ -134,6 +141,29 @@ class MessagingService : FirebaseMessagingService() {
         }
     }
 
+    private fun speakNotification(title: String?) {
+        var textToSpeech: TextToSpeech? = null
+        var tts = title
+        if (tts.isNullOrEmpty())
+            tts = getString(R.string.tts_no_title)
+        textToSpeech = TextToSpeech(applicationContext
+        ) {
+            if (it == TextToSpeech.SUCCESS) {
+                textToSpeech?.speak(tts, TextToSpeech.QUEUE_ADD, null, "")
+                Log.d(TAG, "speaking notification")
+
+                // We need to use some sort of delay to properly shut down the TTS engine
+                // We will be in the background and we don't call onDestroy here
+                // Without this delay nothing will be said and its probably not a good idea to keep it open
+                Thread.sleep(10000)
+                textToSpeech?.stop()
+                textToSpeech?.shutdown()
```

Comment:
Rather than doing this we should use [a listener](https://developer.android.com/reference/android/speech/tts/TextToSpeech#setOnUtteranceProgressListener(android.speech.tts.UtteranceProgressListener)) to determine when we are done so we can shutdown then.